src/core/ui/components/drop-in/drop-down.tsx

```diff
@@ -0,0 +1,414 @@
+import { useUser } from '../../context/user';
+import { useMemo, useState } from 'react';
+import { useSlashAuth } from '../../../context/legacy-slashauth';
+import { ScaleLoader } from 'react-spinners';
+
+export const DropDown = () => {
+  const user = useUser();
+  const [isOpen, setIsOpen] = useState(false);
+  const context = useSlashAuth();
+  const {
+    initialized,
+    logout,
+    loginNoRedirectNoPopup,
+    connect,
+    goToAccountPage,
+  } = context;
+  const [hash, hashDisplay] = useMemo(() => {
+    if (user.wallet) {
+      const hash: string = user.wallet.default.split(':')[1]; //undefined possible
+      const hashDisplay =
```

Comment:
There's a function to shorten wallets here: https://github.com/slashauth/slashauth-react/blob/db1f49fd83470e109c81c73a9d573f2082e4e2a6/src/shared/utils/eth.ts

Reply:
updated